File: 2_prime_generator.cpp
#include <iostream>

using namespace std;


int main()
{
    const int prime_len = 32000;
    bool prime[prime_len];
    for(int i = 0; i < prime_len; ++i)
        prime[i] = true;
    prime[0] = prime[1] = false;

    for(int i = 2; i * i <= prime_len; ++i)
    {
        if(!prime[i])
            continue;
        for(int j = i * i; j <= prime_len; j += i)
            prime[j] = false;
    }
    int m, n;
    int t;
    cin >> t;
    for(int iter  = 0; iter < t; ++iter)
    {
        const int len = 100001;
        cin >> m >> n;
        bool be_prime[len];
        for(int i = 0; i < len; ++i)
            be_prime[i] = true;
        for(int i = 2; i < prime_len; ++i)
        {
            if(i * i > n)
                break;
            if(!prime[i])
                continue;
            
            int start = i - m % i;
            if(start % i == 0)
                start = 0;
            if((start + m) / i == 1)
                start += i;
            for(int j = start; j <= len ; j += i)
            {
                
                be_prime[j] = false;
            }
                
        }
        for(int i = 0 ; i <= n - m ; ++i)
            if(be_prime[i])
                if (i+m != 1)
                cout << i + m << endl;

    }
    
}

File: 2_prime_generator.py

prime_len = 32000
prime_tmp = [True for i in range(prime_len)]
prime_tmp[0] = prime_tmp[1] = False
from math import sqrt

for i in range(2, int(sqrt(prime_len)) + 1):
    if prime_tmp[i]:
        for j in range(i*i, prime_len, i):
            prime_tmp[j] = False
prime = [i for i in range(prime_len) if prime_tmp[i]]

t = int(input())
for i in range(t):
    test_len = 100001
    line = input()
    m, n = line.split()
    m, n = int(m), int(n)
    be_prime = [True for i in range(test_len)]
    for p in prime:
        if p * p > n:
            break
        start = p - m % p
        if start % p == 0:
            start = 0
        if (start + m) // p == 1:
            start += p
        for j in range(start, test_len, p):
            be_prime[j] = False
    for i in range(n-m+1):
        if be_prime[i]:
            if i + m != 1:
                print (i + m)
        